frontend/packages/console-app/src/redux/reducers/guided-tour-reducer.ts

```diff
@@ -0,0 +1,40 @@
+import { GuidedTourSidebarActions, Actions } from '../actions/guided-tour-actions';
+import { RootState } from '@console/internal/redux';
+import { GuidedTourStatus } from '../../components/guided-tours/utils/guided-tour-status';
+
+export const guidedTourReducerName = 'guidedTour';
+
+type State = {
+  isExpanded: boolean;
+  activeTourID: string;
+};
+
+const initialState: State = {
+  isExpanded: false,
+  activeTourID: '',
+};
+
+export default (state = initialState, action: GuidedTourSidebarActions) => {
+  if (action.type === Actions.ToggleGuidedTourSidebarExpanded) {
+    return {
+      isExpanded:
+        state.isExpanded &&
+        action.payload.activeTourID &&
+        action.payload.activeTourID !== state.activeTourID
+          ? state.isExpanded
+          : !state.isExpanded,
```

Comment:
used activeTourID to toggle expand collapse of drawer
but that causes the contents of the sidebar to be undefined while the sidebar is closing, which looks odd
![drawercontentundefined](https://user-images.githubusercontent.com/38663217/86921399-b5324c80-c148-11ea-9a55-23349ce423cb.gif)

Reply:
This would have been the case with your original change as well. I think this can be fixed using the following:

```
isExpanded: !!action.payload.activeTourID,
activeTourID: action.payload.activeTourID || state. activeTourID
```